biblib/views/operations_view.py

```diff
@@ -207,6 +207,10 @@ def post(self, library):
         if data['action'] in ['union', 'intersection', 'difference']:
             if 'libraries' not in data:
                 return err(NO_LIBRARY_SPECIFIED_ERROR)
+            for lib in data['libraries']:
+                if not self.read_access(service_uid=user_editing_uid,
```

Comment:
This should be a check on if the user has read access, or if the library is public

Reply:
Just added the access check when library is public. A slug is passed for each library, so we have to convert it to UUID and then get the library to check if it's public. I noticed there was code right below that sort of did that, so I moved some things around.